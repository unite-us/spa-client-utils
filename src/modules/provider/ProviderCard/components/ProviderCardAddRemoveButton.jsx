import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { noop } from 'lodash';
import AddRemoveButtons from 'modules/AddRemoveButtons';

class ProviderCardAddRemoveButton extends Component {
  constructor(props) {
    super(props);

    this.providerCardAdd = this.providerCardAdd.bind(this);
  }

  providerCardAdd() {
    const { destinationLatLng, onAddClick } = this.props;

    onAddClick(destinationLatLng);
  }

  render() {
    const {
      className,
      onRemoveClick,
      selected,
    } = this.props;

    return (
      <div
        className={
          classnames('provider-card-add-remove-button', className)
        }
      >
        <AddRemoveButtons
          index={selected ? 2 : 0}
          length={selected ? 2 : 1}
          onAddClick={this.providerCardAdd}
          onRemoveClick={onRemoveClick}
        />
      </div>
    );
  }
}

ProviderCardAddRemoveButton.propTypes = {
  className: PropTypes.string,
  destinationLatLng: PropTypes.array,
  onAddClick: PropTypes.func,
  onRemoveClick: PropTypes.func,
  selected: PropTypes.bool,
};

ProviderCardAddRemoveButton.defaultProps = {
  className: '',
  destinationLatLng: [],
  onAddClick: noop,
  onRemoveClick: noop,
  selected: false,
};

export default ProviderCardAddRemoveButton;
