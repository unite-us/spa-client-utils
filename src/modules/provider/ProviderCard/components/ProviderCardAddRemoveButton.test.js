import React from 'react';
import renderer from 'react-test-renderer';
import ProviderCardAddRemoveButton from './ProviderCardAddRemoveButton';

jest.mock('@unite-us/ui', () => ({
  ...require.requireActual('@unite-us/ui'),
  Icon: jest.fn(() => <mock-icon />),
}));

describe('ProviderCardAddRemoveButton', () => {
  it('renders add button', () => {
    expect(renderer.create(
      <ProviderCardAddRemoveButton className="another-class-name" />,
    )).toMatchSnapshot();
  });

  it('renders remove button', () => {
    expect(renderer.create(
      <ProviderCardAddRemoveButton
        className="another-class-name"
        selected
      />,
    )).toMatchSnapshot();
  });
});
