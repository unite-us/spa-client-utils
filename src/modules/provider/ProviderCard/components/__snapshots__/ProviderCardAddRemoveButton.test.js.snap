// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProviderCardAddRemoveButton renders add button 1`] = `
<div
  className="provider-card-add-remove-button another-class-name"
>
  <div
    className="ui-add-remove-buttons"
  >
    <div
      className="ui-add-remove-buttons__add"
    >
      <a
        onClick={[Function]}
        role="button"
        tabIndex={0}
        title="Add"
      >
        <mock-icon />
      </a>
    </div>
  </div>
</div>
`;

exports[`ProviderCardAddRemoveButton renders remove button 1`] = `
<div
  className="provider-card-add-remove-button another-class-name"
>
  <div
    className="ui-add-remove-buttons"
  >
    <div
      className="ui-add-remove-buttons__remove mr-quarter"
    >
      <a
        onClick={[Function]}
        role="button"
        tabIndex={0}
        title="Remove"
      >
        <mock-icon />
      </a>
    </div>
  </div>
</div>
`;
